Mathlib/Tactic/Linter/TextBased.lean

```diff
@@ -338,9 +350,39 @@ def broadImportsLinter : TextbasedLinter := fun lines ↦ Id.run do
             else if name == "Lake" || name.startsWith "Lake." then
               errors := errors.push (StyleError.broadImport BroadImports.Lake, lineNumber)
       lineNumber := lineNumber + 1
-  return errors
+  return (errors, none)
 
 
+/-- Lint a collection of input strings if one of them contains windows line endings. -/
+def windowsLineEndingLinter : TextbasedLinter := fun lines ↦ Id.run do
+  let mut errors := Array.mkEmpty 0
+  let mut fixedLines := lines
+  -- invariant: this equals the current index of 'line' in 'lines',
+  -- hence starts at 0 and is incremented *at the end* of the loop
+  let mut lineNumber := 0
+  for line in lines do
+    let replaced := line.crlfToLf
+    if replaced != line then
+      errors := errors.push (StyleError.windowsLineEnding, lineNumber)
+      fixedLines := fixedLines.set! 0 replaced
```

Comment:
```suggestion
      fixedLines := fixedLines.set! lineNumber replaced
```
If I am following correctly, you are replacing the `lineNumber`th entry of `lines`, not constantly updating the `0`th one, right?

Reply:
Indeed, that was a typo/logic error. Good catch!